src/wallet/mod.rs

```diff
@@ -2557,7 +2569,9 @@ pub(crate) mod test {
         let addr = Address::from_str("2N1Ffz3WaNzbeLFBb51xyFMHYSEUXcbiSoX").unwrap();
         let utxo = wallet2.list_unspent().unwrap().remove(0);
         let foreign_utxo_satisfaction = wallet2
-            .get_descriptor_for_keychain(KeychainKind::External)
+            .get_descriptor_for_script_pubkey(&utxo.txout.script_pubkey)
+            .unwrap()
+            .unwrap()
```

Comment:
Replacing `get_descriptor_for_keychain` with `get_descriptor_for_script_pubkey` should no longer be necessary.

Revert these.